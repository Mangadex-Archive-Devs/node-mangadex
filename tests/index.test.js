const mangadex = require(`${__dirname}/../index`)
const chapterList = require(`${__dirname}/files/chapterList.json`)


describe('Endpoints', () => {
    test('getManga()', async () => {
        expect.assertions(1)
        expect(await mangadex.getManga(6510)).toMatchSnapshot()
    })
    test('getChapter()', async () => {
        expect(await mangadex.getChapter(45522)).toMatchSnapshot({
            server: expect.any(String),
        })
    })
})

describe('Internals', () => {
    describe('areChaptersSequential', () => {
        /*
            map:
        */
       function newChapterObject(str) {
           const regex = new RegExp(/v(.+)c(.+)/)
           const results = regex.exec(str)

           return {
               volume: results[1] !== '?' ? results[1] : '',
               chapter: results[2] !== '?' ? results[2] : ''
           }
       }

        describe('Ascending', () => {
            test('v1c1 & v1c2', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c1'),
                        newChapterObject('v1c2')
                    )
                )
                    .toBe(true)
            })
            test('v1c1 & v?c2', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c1'),
                        newChapterObject('v?c2')
                    )
                )
                    .toBe(true)
            })
            test('v1c1 & v2c6', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c1'),
                        newChapterObject('v2c6')
                    )
                )
                    .toBe(false)
            })
            test('v1c1 & v1c?', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c1'),
                        newChapterObject('v1c?')
                    )
                )
                    .toBe(undefined)
            })
            test('v1c1.1 & v1c1.2', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c1.1'),
                        newChapterObject('v1c1.2')
                    )
                )
                    .toBe(true)
            })
            test('v1c1.2 & v2c1.1', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c1.2'),
                        newChapterObject('v2c1.1')
                    )
                )
                    .toBe(true)
            })
            test('v1c1.2 & v2c6', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c1.2'),
                        newChapterObject('v2c6')
                    )
                )
                    .toBe(false)
            })
            test('v1c5 & v2c1', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c5'),
                        newChapterObject('v2c1')
                    )
                )
                    .toBe(true)
            })
            test('v1c1 & v2c3', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c1'),
                        newChapterObject('v2c3')
                    )
                )
                    .toBe(false)
            })
            test('v1c5 & v3c1', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c5'),
                        newChapterObject('v3c1')
                    )
                )
                    .toBe(false)
            })
            test('v1c5 & v2c1.1', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c5'),
                        newChapterObject('v2c1.1')
                    )
                )
                    .toBe(true)
            })
            test('v1c5 & v2c1.2', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c5'),
                        newChapterObject('v2c1.2')
                    )
                )
                    .toBe(false)
            })
        })

        describe('Descending', () => {
            test('v1c2 & v1c1', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c2'),
                        newChapterObject('v1c1')
                    )
                )
                    .toBe(true)
            })
            test('v?c2 & v1c1', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v?c2'),
                        newChapterObject('v1c1')
                    )
                )
                    .toBe(true)
            })
            test('v2c6 & v1c1', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v2c6'),
                        newChapterObject('v1c1')
                    )
                )
                    .toBe(false)
            })
            test('v1c? & v1c1', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c?'),
                        newChapterObject('v1c1')
                    )
                )
                    .toBe(undefined)
            })
            test('v1c1.2 & v1c1.1', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v1c1.2'),
                        newChapterObject('v1c1.1')
                    )
                )
                    .toBe(true)
            })
            test('v2c1.1 & v1c1.2', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v2c1.1'),
                        newChapterObject('v1c1.2')
                    )
                )
                    .toBe(true)
            })
            test('v2c6 & v1c1.2', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v2c6'),
                        newChapterObject('v1c1.2')
                    )
                )
                    .toBe(false)
            })

            test('v2c1 & v1c5', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v2c1'),
                        newChapterObject('v1c5')
                    )
                )
                    .toBe(true)
            })
            test('v2c3 & v1c1', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v2c3'),
                        newChapterObject('v1c1')
                    )
                )
                    .toBe(false)
            })
            test('v3c1 & v1c5', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v3c1'),
                        newChapterObject('v1c5')
                    )
                )
                    .toBe(false)
            })
            test('v2c1.1 & v1c5', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v2c1.1'),
                        newChapterObject('v1c5')
                    )
                )
                    .toBe(true)
            })
            test('v2c1.2 & v1c5', () => {
                expect(
                    mangadex.internals.areChaptersSequential(
                        newChapterObject('v2c1.2'),
                        newChapterObject('v1c5')
                    )
                )
                    .toBe(false)
            })
        })
    })
    describe('sortChapters', () => {
        test('Just Works (tm)', () => {
            expect(mangadex.internals.sortChapters(chapterList))
                .toMatchSnapshot()
        })
    })
})