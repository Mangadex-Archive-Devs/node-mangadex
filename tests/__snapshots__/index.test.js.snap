// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Endpoints getChapter() 1`] = `
Object {
  "chapter": "1",
  "comments": null,
  "group_id": 2,
  "group_id_2": 0,
  "group_id_3": 0,
  "hash": "26eea9c2123d7cc63577de694f0b72d1",
  "id": 45522,
  "lang_code": "gb",
  "lang_name": "English",
  "long_strip": 0,
  "manga_id": 6510,
  "page_array": Array [
    "x1.jpg",
    "x2.jpg",
    "x3.jpg",
    "x4.jpg",
    "x5.jpg",
    "x6.jpg",
    "x7.jpg",
    "x8.jpg",
    "x9.jpg",
    "x10.jpg",
    "x11.jpg",
    "x12.jpg",
    "x13.jpg",
  ],
  "server": Any<String>,
  "status": "OK",
  "timestamp": 1517858262,
  "title": "Episode Minus 1",
  "volume": "",
}
`;

exports[`Endpoints getManga() 1`] = `
Object {
  "chapter": undefined,
  "chapters": Array [
    Object {
      "chapter": "1",
      "group_id": 2,
      "group_id_2": 0,
      "group_id_3": 0,
      "group_name": "Unknown",
      "group_name_2": null,
      "group_name_3": null,
      "id": 45522,
      "lang_code": "gb",
      "timestamp": 1517858262,
      "title": "Episode Minus 1",
      "volume": "",
    },
    Object {
      "chapter": "2",
      "group_id": 2,
      "group_id_2": 0,
      "group_id_3": 0,
      "group_name": "Unknown",
      "group_name_2": null,
      "group_name_3": null,
      "id": 45524,
      "lang_code": "gb",
      "timestamp": 1517858281,
      "title": "Fractional Episode",
      "volume": "",
    },
    Object {
      "chapter": "3",
      "group_id": 2,
      "group_id_2": 0,
      "group_id_3": 0,
      "group_name": "Unknown",
      "group_name_2": null,
      "group_name_3": null,
      "id": 45526,
      "lang_code": "gb",
      "timestamp": 1517858296,
      "title": "Special Episode",
      "volume": "",
    },
  ],
  "manga": Object {
    "artist": "Azuma Kiyohiko",
    "author": "Azuma Kiyohiko",
    "cover_url": "/images/manga/6510.jpg?1519173421",
    "description": "Kiyohiko Azuma's Try! Try! Try! follows the adventures of a weird girl named Yotsuba whose bizarre antics baffle her friends and her father. The characters in this comic later went on to appear in the monthly Yotsubato! serial manga currently running in Dengeki Daioh magazine.",
    "genres": Array [
      5,
      31,
    ],
    "lang_flag": "jp",
    "lang_name": "Japanese",
    "last_chapter": "0",
    "status": 2,
    "title": "Try! Try! Try!",
  },
  "status": "OK",
}
`;

exports[`Internals sortChapters Just Works (tm) 1`] = `
Array [
  Object {
    "chapter": "",
    "group_id": "3478",
    "volume": "1",
  },
  Object {
    "chapter": "1",
    "group_id": "3043",
    "volume": "1",
  },
  Object {
    "chapter": "1",
    "group_id": "3478",
    "volume": "1",
  },
  Object {
    "chapter": "1.1",
    "group_id": "3478",
    "volume": "1",
  },
  Object {
    "chapter": "1.2",
    "group_id": "3478",
    "volume": "1",
  },
  Object {
    "chapter": "2",
    "group_id": "3043",
    "volume": "1",
  },
  Object {
    "chapter": "2",
    "group_id": "3478",
    "volume": "",
  },
  Object {
    "chapter": "1.1",
    "group_id": "3478",
    "volume": "2",
  },
  Object {
    "chapter": "6",
    "group_id": "3043",
    "volume": "2",
  },
]
`;
